test/number-literal-case.js

```diff
@@ -0,0 +1,75 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/number-literal-case';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	},
+	parserOptions: {
+		sourceType: 'module'
+	}
+});
+
+const error = {
+	ruleId: 'number-literal-case',
+	message: 'Invalid number literal casing'
+};
+
+ruleTester.run('number-literal-case', rule, {
+	valid: [
+		'const foo = 0xFF',
+		'const foo = 0b11',
+		'const foo = 0o10',
+		`const foo = '0Xff'`
```

Comment:
@jfmengels It's a string that I test here. When I created the rule, I was on my bike thinkering about this "should I explicitely test if the literal is of type number". This is not needed as the `raw` value of a string is `"'0Xff'"`. With this test, I just wanted to make sure this is not enforced on string values. Does it make sense somehow :p?